kyaml/kio/byteio_reader.go

```diff
@@ -119,7 +123,20 @@ func (r *ByteReader) Read() ([]*yaml.RNode, error) {
 
 	index := 0
 	for i := range values {
-		decoder := yaml.NewDecoder(bytes.NewBufferString(values[i]))
+		value := values[i]
+
+		for _, g := range JSONMatch {
+			if match, err := filepath.Match(g, filepath.Ext(r.Path)); err != nil {
```

Comment:
why not just `g == filepath.Ext(r.Path)` ?

Reply:
As filepath.Ext(r.Path) is .json and g is *.json.

Reply:
Removed the filematch logic.